plugin/core/registry.py

```diff
@@ -161,3 +168,143 @@ def run_async() -> None:
         sublime.set_timeout_async(lambda: windows.lookup(self.window).restart_sessions_async(config_name))
+
+
+# The server capabilities which are possible to toggle on/off.
+# This should only contain the capabilities which are related to a certain feature.
+# https://microsoft.github.io/language-server-protocol/specifications/specification-current/#serverCapabilities
+SERVER_CAPABILITIES = {
+    "completionProvider": "Auto Complete",
+    "hoverProvider": "Hover",
+    "signatureHelpProvider": "Signature Help",
+    "declarationProvider": "Goto Declaration",
+    "definitionProvider": "Goto Definition",
+    "typeDefinitionProvider": "Goto Type Definition",
+    "implementationProvider": "Goto Implementation",
+    "referencesProvider": "Find References",
+    "documentHighlightProvider": "Highlights",
+    "documentSymbolProvider": "Goto Symbol",
+    "codeActionProvider": "Code Actions",
+    "codeLensProvider": "Code Lenses",
+    "documentLinkProvider": "Links",
+    "colorProvider": "Color Boxes",
+    "documentFormattingProvider": "Formatting",
+    "documentRangeFormattingProvider": "Range Formatting",
+    # "documentOnTypeFormattingProvider": "On Type Formatting",  # not supported by this client
+    "renameProvider": "Rename",
+    # "foldingRangeProvider": "Folding",  # not supported by this client
+    "executeCommandProvider": "Run Server Command",
+    "selectionRangeProvider": "Expand Selection",
+    # "linkedEditingRangeProvider": "Linked Editing",  # not supported by this client
+    # "callHierarchyProvider": "Call Hierarchy",  # not supported by this client
+    "semanticTokensProvider": "Semantic Highlighting",
+    # "typeHierarchyProvider": "Type Hierarchy",  # not supported by this client
+    "inlayHintProvider": "Inlay Hints",
+    # "diagnosticProvider": "Pull Diagnostics",  # not supported by this client
+    "workspaceSymbolProvider": "Goto Symbol in Project"
+}
+
+
+class LspToggleCapabilityCommand(LspWindowCommand):
+
+    last_toggled = None
+
+    def run(self, capability: str) -> None:
+        global toggled_capabilities
+        if capability not in SERVER_CAPABILITIES:
+            raise ValueError("Invalid capability name: {}".format(capability))
+        if capability in toggled_capabilities:
+            toggled_capabilities.remove(capability)
+            new_state = "on"
+        else:
+            toggled_capabilities.add(capability)
+            new_state = "off"
+        self.last_toggled = capability
+        if capability == "hoverProvider":
+            user_setting = sublime.load_settings("Preferences.sublime-settings").get("show_definitions")
+            for session in self.sessions():
+                for sv in session.session_views_async():
+                    if new_state == "on":
+                        sv.view.settings().set("show_definitions", False)
+                    else:
+                        sv.view.settings().set("show_definitions", user_setting)
```

Comment:
Haven't investigated but after disabling `hoverProvider` for LSP-typescript on a document that runs LSP-eslint and LSP-typescript I'm getting weird flash of content that immediately disappears when showing the popup.

I could investigate if we're gonna proceed with this.